src/st/no3/3FF00.c

```diff
@@ -0,0 +1,1208 @@
+#include "no3.h"
+
+void EntityRoomTransition2(Entity* self) {
+    Entity* newEntity;
+    Primitive* prim;
+    // Used as both a loop variable and a temp
+    s32 localVar;
+
+    // These are weird fake vars needed for matching
+    Entity* gents = g_Entities;
+    Unkstruct8* layout = &g_CurrentRoomTileLayout;
+
+    if (self->ext.generic.unk80.modeS32 == 0 && self->step < 4) {
+        g_api.PlaySfx(0x631);
+        self->ext.generic.unk80.modeS32 = 0x200;
+    }
+    self->ext.generic.unk80.modeS32--;
+
+    switch (self->step) {
+    case 0:
+        InitializeEntity(D_80180AD0);
+        layout->unk40 = 0xFC;
+        g_Player.D_80072EF4 = 0x2000;
+        D_8003C8B8 = 0;
+        g_Player.D_80072EFC = 1;
+        D_801D7DD0 |= 0x100;
+        return;
+    case 1:
+        // Evil use of local 'gents' instead of PLAYER
+        if (gents[0].posX.i.hi < 0x34) {
+            g_Entities[1].ext.generic.unk7C.S8.unk0 = 1;
+            g_Player.D_80072EF4 = 0x2000;
+        } else {
+            g_Player.D_80072EF4 = 0;
+            D_801D7DD0 |= 0x80;
+            self->step++;
+        }
+        g_Player.D_80072EFC = 1;
+        return;
+    case 2:
+        if (D_801D7DD0 & 0x20) {
+            g_api.func_800FF7B8(1);
+            g_api.PlaySfx(0x6A2);
+            for (localVar = 0; localVar < 6; localVar++) {
+                newEntity = AllocEntity(&g_Entities[224], &g_Entities[256]);
+                if (newEntity == NULL) {
+                    break;
+                }
+                CreateEntityFromEntity(0x59, gents, newEntity);
+                newEntity->params = localVar;
+            }
+            g_Player.D_80072EF4 = 0x10000;
+            self->ext.generic.unk7C.s = 0x10;
+            self->step++;
+        }
+        g_Player.D_80072EFC = 1;
+        return;
+    case 3:
+        if (D_801D7DD0 & 0x40 && !(--self->ext.generic.unk7C.s & 0xFFFF)) {
+            localVar = g_api.AllocPrimitives(PRIM_TILE, 1);
+            if (localVar != -1) {
+                prim = &g_PrimBuf[localVar];
+                self->primIndex = localVar;
+                self->flags |= 0x800000;
+                prim->u0 = 0xFF;
+                prim->v0 = 0xFF;
+                prim->priority = 0x50;
+                prim->x0 = 0;
+                prim->y0 = 0;
+                prim->g0 = 0;
+                prim->b0 = 0;
+                prim->r0 = 0;
```

Comment:
```suggestion
prim->r0 = prim->b0 = prim->g0 = prim->y0 = prim->x0 = 0;

Reply:
While I agree on `prim->r0 = prim->b0 = prim->g0 = 0;` (it is a pattern commonly seen in the decomp) I do not for `prim->y0 = prim->x0 = 0;`. The struct field order should be from the left to the right and from the top to the bottom whenever it is matching regardless the order. This is also valid for the `prim->v0 = prim->u0 = 0xFF;` suggested/requested below.